pkg/storaged/btrfs/subvolume.jsx

```diff
@@ -206,6 +297,17 @@ export function make_btrfs_subvolume_page(parent, volume, subvol) {
         action: () => subvolume_create(volume, subvol, (mounted && !opt_ro) ? mount_point : mount_point_in_parent),
     });
 
+    let delete_excuse = create_excuse;
+    if (subvol.id === 5)
+        delete_excuse = _("Deleting the root subvolume is not allowed");
+
+    actions.push({
+        danger: true,
```

Comment:
Somehow a disabled danger action doesn't look disabled:

![image](https://github.com/cockpit-project/cockpit/assets/3228183/b939c7ab-6cc0-4b8d-810b-372a913e3249)

I guess we need to change that in pages.jsx.

Reply:
Should it auto-mount to do the action, when that's needed?

And to delete a subvolume, doesn't the parent volume need to be mounted, not the subvolume itself? (I might be wrong.)

If we want to let people know it needs to be mounted, we could have it enabled and say something like:

Create subvolume
Volume will be mounted to create a subvolume

Then the question is if it should return to the original state (or stay mounted even if it wasn't)

Reply:
It might not have a mount point so we can't easily figure that out and that feels a bit too magic for me.

Reply:
> Should it auto-mount to do the action, when that's needed?

Good question. We could mount it temporarily behind the scenes in some random place in /tmp/, for example.  This would be similar to resizing other filesystems: some filesystems need to be mounted to be resized, and when they are not, libblockdev will temporarily mount them in a secret location.

Reply:
> And to delete a subvolume, doesn't the parent volume need to be mounted, not the subvolume itself? (I might be wrong.)

You are correct.

Reply:
> Create subvolume Volume will be mounted to create a subvolume
> 
> Then the question is if it should return to the original state (or stay mounted even if it wasn't)

This assumes that the subvolume already has a assigned mount point, right? Or would we open the usual "Mount" dialog?

Reply:
> We would need to do this for deletion and creation

Also for listing?

Reply:
Made a separate PR for that https://github.com/cockpit-project/cockpit/pull/19898